pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -93,63 +105,21 @@ func machinesEqual(m1 *clusterv1.Machine, m2 *clusterv1.Machine) bool {
 	return m1.Name == m2.Name && m1.Namespace == m2.Namespace
 }
 
-// isNodeJoin determines if a machine, in scope, should join of the cluster.
-// TODO: Make this thread safe kubernetes-sigs/cluster-api#925
-// https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/745#discussion_r280506890
-func (a *Actuator) isNodeJoin(scope *actuators.MachineScope, controlPlaneMachines []*clusterv1.Machine) (bool, error) {
-	switch set := scope.Machine.ObjectMeta.Labels["set"]; set {
-	case "node":
-		// Worker machines, not part of the controlplane, will always join the cluster.
-		return true, nil
-	case "controlplane":
-		// Controlplane machines will join the cluster if the cluster has an existing control plane.
-		controlplaneExists := false
-		var err error
-
-		var sharedScope *actuators.MachineScope
-
-		for _, cm := range controlPlaneMachines {
-			if sharedScope == nil {
-				sharedScope, err = actuators.NewMachineScope(actuators.MachineScopeParams{
-					Machine:    cm,
-					Cluster:    scope.Cluster,
-					Client:     a.clusterClient,
-					Logger:     a.log,
-					AWSClients: scope.AWSClients,
-				})
-				if err != nil {
-					return false, errors.Wrapf(err, "failed to create machine scope for %s/%s", cm.Namespace, cm.Name)
-				}
-			}
-			sharedScope.Machine = cm
-			ec2svc := ec2.NewService(sharedScope.Scope)
-
-			controlplaneExists, err = ec2svc.MachineExists(sharedScope)
-			if err != nil {
-				return false, errors.Wrapf(err, "failed to verify existence of machine %s/%s", sharedScope.Machine.Namespace, sharedScope.Machine.Name)
-			}
-
-			if controlplaneExists {
-				return controlplaneExists, err
-			}
-		}
-
-		a.log.V(2).Info("Will machine join the controlplane", "machine-name", scope.Machine.Name, "machine-namespace", scope.Machine.Name, "should-join-control-plane", controlplaneExists)
-		return controlplaneExists, err
-
-	default:
-		return false, errors.Errorf("Unknown value %q for label `set` on machine %q", set, scope.Machine.Name)
-	}
-}
-
 // Create creates a machine and is invoked by the machine controller.
 func (a *Actuator) Create(ctx context.Context, cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
 	if cluster == nil {
 		return errors.Errorf("missing cluster for machine %s/%s", machine.Namespace, machine.Name)
 	}
-	a.log.Info("Creating machine in cluster", "machine-name", machine.Name, "machine-namespace", machine.Namespace, "cluster-name", cluster.Name)
 
-	scope, err := actuators.NewMachineScope(actuators.MachineScopeParams{Machine: machine, Cluster: cluster, Client: a.clusterClient, Logger: a.log})
+	log := a.log.WithValues("machine-name", machine.Name, "namespace", machine.Namespace, "cluster-name", cluster.Name)
+	log.Info("Processing machine creation")
+
+	if cluster.Annotations[v1alpha1.AnnotationClusterInfrastructureReady] != v1alpha1.ValueReady {
```

Comment:
So the cluster actuator, to set the cluster-infra, waits for a control-plane machine to have a nodeRef which will happen once the machine is created. However, seems like the machine actuator also waits for the cluster infra to be ready. Am I missing something?

Reply:
There are 2 different annotations: infra-ready, and control-plane-ready. The cluster actuator waits for the infra (VPC, ELB, etc) to be ready, then it sets infra-ready. It does not wait for the control plane for infra-ready.

Reply:
Yes, ofcourse I was missing something. The name of the annotations `AnnotationClusterInfrastructureReady` and `AnnotationControlPlaneReady`

Reply:
🤦‍♂